src/react/src/components/Dashboard/DashboardModerationQueue.jsx

```diff
@@ -112,41 +124,57 @@ const DashboardModerationQueue = ({
     }, [dataSources, moderationStatuses, countries]);
 
     const handleAfterDateChange = date => {
-        if (!beforeDate || !date || date <= beforeDate) {
-            setAfterDateError(false);
-            dispatch(updateAfterDate(date));
-            dispatch(clearModerationEvents());
-            dispatch(
-                updateModerationEventsPage({
-                    page: MODERATION_INITIAL_PAGE_INDEX,
-                    maxPage: MODERATION_INITIAL_PAGE_INDEX,
-                    pageSize,
-                }),
-            );
-            fetchEvents();
-        } else {
+        if (!isValidDate(date)) {
             dispatch(updateAfterDate(''));
+            setErrorDateText(DATE_FORMAT_ERROR);
             setAfterDateError(true);
+            return;
         }
+        if (date && beforeDate && date > beforeDate) {
+            dispatch(updateAfterDate(''));
+            setErrorDateText(DATE_RANGE_ERROR);
+            setAfterDateError(true);
+            return;
+        }
+
+        setAfterDateError(false);
+        dispatch(updateAfterDate(date));
+        dispatch(clearModerationEvents());
+        dispatch(
+            updateModerationEventsPage({
+                page: MODERATION_INITIAL_PAGE_INDEX,
+                maxPage: MODERATION_INITIAL_PAGE_INDEX,
+                pageSize,
+            }),
+        );
+        fetchEvents();
     };
 
     const handleBeforeDateChange = date => {
-        if (!afterDate || !date || date >= afterDate) {
-            setBeforeDateError(false);
-            dispatch(updateBeforeDate(date));
-            dispatch(clearModerationEvents());
-            dispatch(
-                updateModerationEventsPage({
-                    page: MODERATION_INITIAL_PAGE_INDEX,
-                    maxPage: MODERATION_INITIAL_PAGE_INDEX,
-                    pageSize,
-                }),
-            );
-            fetchEvents();
-        } else {
+        if (!isValidDate(date)) {
+            dispatch(updateBeforeDate(''));
+            setErrorDateText(DATE_FORMAT_ERROR);
+            setBeforeDateError(true);
+            return;
+        }
+        if (date && afterDate && date < afterDate) {
```

Comment:
I've noticed double usage of this check. What about to move it into a separate function?

Reply:
Done.

Reply:
Also a litle bit reworked due to requirements